src/solver/type_getters.jl

```diff
@@ -661,6 +666,80 @@ function get_simulation(config::AtmosConfig)
 
     tracers = get_tracers(config.parsed_args)
 
+    if initial_condition isa ClimaAtmos.InitialConditions.DYAMONDSummer
+        @show "DYAMONDSummer, initialize from file"
+        thermo_params = params.thermodynamics_params
+        file_path = AA.dyamond_summer_artifact_path(;
+            context = get_comms_context(config.parsed_args),
+        )
+        Pressure = ClimaUtilities.SpaceVaryingInputs.SpaceVaryingInput(
+            file_path,
+            "p",
+            spaces.center_space,
+        )
+        Temperature = ClimaUtilities.SpaceVaryingInputs.SpaceVaryingInput(
+            file_path,
+            "t",
+            spaces.center_space,
+        )
+        q_tot = ClimaUtilities.SpaceVaryingInputs.SpaceVaryingInput(
+            file_path,
+            "q",
+            spaces.center_space,
+        )
+        q = TD.PhasePartition.(q_tot)
+        Y.c.ρ .= TD.air_density.(thermo_params, Temperature, Pressure, q)
```

Comment:
As far as possible let's try to construct `thermo_states` based on known properties, then, in any call to `Thermodynamics` we can pass this state as an arg and ensure that our vars are used consistently.

Reply:
Yes, this is better, and can reduce potential bugs as the one above.